### ISSUE TITLE
Wrong Date selection

### ISSUE THREAD

**Issue by ravpninfotech** [peterent/MultiDatePicker#1]
Get the previous date when I'm select a particular date.

For example: if I'm select the date as of 05/04/2021 but when I print the "dayOfMonth.date" then it will give me the date is 04/04/2021.
I had invested more than 2 days for this issue but there is no luck could you please help me with that 
thanks in advance

**Comment by peterent** [comment]
The month index begins at zero, not one. Looks like its just a matter of adding 1 to the value you are getting. Otherwise please post some code showing the issue.

**Comment by ravpninfotech** [comment]
> The month index begins at zero, not one. Looks like its just a matter of adding 1 to the value you are getting. Otherwise please post some code showing the issue.

struct MDPDayView: View  {

 private func handleSelection() {
        if dayOfMonth.isSelectable {
            monthDataModel.selectDay(dayOfMonth)
        }
    }
    
    var body: some View {
        Button( action: {handleSelection()} ) {
            Text("\(dayOfMonth.day)")
                .font(.headline)
                .fontWeight(.medium)
                .foregroundColor(textColor)
                .frame(minHeight: cellSize, maxHeight: cellSize)
                .background(
                    Circle()
                        .stroke(strokeColor, lineWidth: 1)
                        .background(Circle().foregroundColor(fillColor))
                        .frame(width: cellSize, height: cellSize)
                )
        }.foregroundColor(.black)
    }
 }
 
- Selected date fill color is right but gives me the wrong date.

**Comment by peterent** [comment]
So this will be a hack fix for you. I'd have to go back and figure out the best approach to make it seamless, but this should work for now.

In your handleSelection you know the month component is off by 1. Given a date, you can bump it forward by a month using this:

let newDate = Calendar.current.date(byAdding: .month, value: 1, to: date)

If you can bear with me, I will try and sort this out soon and make an update, but I didn't want you to wait any longer than you already have.

**Comment by ravpninfotech** [comment]
> So this will be a hack fix for you. I'd have to go back and figure out the best approach to make it seamless, but this should work for now.
> 
> In your handleSelection you know the month component is off by 1. Given a date, you can bump it forward by a month using this:
> 
> let newDate = Calendar.current.date(byAdding: .month, value: 1, to: date)
> 
> If you can bear with me, I will try and sort this out soon and make an update, but I didn't want you to wait any longer than you already have.

Yes, I will waiting for a solution from your side. But please give me a solution ASAP.

**Comment by ravpninfotech** [comment]
> > So this will be a hack fix for you. I'd have to go back and figure out the best approach to make it seamless, but this should work for now.
> > In your handleSelection you know the month component is off by 1. Given a date, you can bump it forward by a month using this:
> > let newDate = Calendar.current.date(byAdding: .month, value: 1, to: date)
> > If you can bear with me, I will try and sort this out soon and make an update, but I didn't want you to wait any longer than you already have.
> 
> Yes, I will waiting for solution from your side.

**Comment by peterent** [comment]
Here's what I'd like you to do to help me figure this out. Since you have this repository, please go into the `ContentView.swift` file and do the following:

1. Change the testMinDate and testMaxDate values so they are before and after the current date so the test will work.

```
let testMinDate = Calendar.current.date(from: DateComponents(year: 2021, month: 4, day: 1))
let testMaxDate = Calendar.current.date(from: DateComponents(year: 2021, month: 5, day: 30))
```

2. Add the following code to the TabView, just below line 99 in that same file:

```
.onChange(of: self.selectedDate, perform: { date in
          print("Single date selected: \(date)")
})
.onChange(of: self.anyDays, perform: { days in
        print("Any days selected: \(days)")
})
.onChange(of: self.dateRange, perform: { dateRange in
        print("Range selected: \(dateRange)")
})
```

3. Run the sample app. The console will print the selected days. When I run it and pick the 3rd of May 2021, I get this in the console:

Single date selected: 2021-05-03 04:00:00 +0000

which is what I would expect to get. It seems like you would be getting something different.

Thank you.
Peter

**Comment by peterent** [comment]
Actually, you can get do a pull from the repo - I pushed this change to it and its a good idea to do the print statements for verification.

**Comment by ravpninfotech** [comment]
> Here's what I'd like you to do to help me figure this out. Since you have this repository, please go into the `ContentView.swift` file and do the following:
> 
> 1. Change the testMinDate and testMaxDate values so they are before and after the current date so the test will work.
> 
> ```
> let testMinDate = Calendar.current.date(from: DateComponents(year: 2021, month: 4, day: 1))
> let testMaxDate = Calendar.current.date(from: DateComponents(year: 2021, month: 5, day: 30))
> ```
> 
> 1. Add the following code to the TabView, just below line 99 in that same file:
> 
> ```
> .onChange(of: self.selectedDate, perform: { date in
>           print("Single date selected: \(date)")
> })
> .onChange(of: self.anyDays, perform: { days in
>         print("Any days selected: \(days)")
> })
> .onChange(of: self.dateRange, perform: { dateRange in
>         print("Range selected: \(dateRange)")
> })
> ```
> 
> 1. Run the sample app. The console will print the selected days. When I run it and pick the 3rd of May 2021, I get this in the console:
> 
> Single date selected: 2021-05-03 04:00:00 +0000
> 
> which is what I would expect to get. It seems like you would be getting something different.
> 
> Thank you.
> Peter


https://user-images.githubusercontent.com/48705441/117006600-4725ca00-ad06-11eb-9307-8c460131c7a7.png
Please check above screenshot. I'm select multiple dates in Any tab "2021-05-04, 2021-05-05" but print different dates in console using your code.
https://user-images.githubusercontent.com/48705441/117007241-08444400-ad07-11eb-888d-a9edeb3988c7.png

So please give me solution.

**Comment by peterent** [comment]
Sorry for the delay again. Been trying to get acclimated to a new job. What timezone are you in? The dates are adjusted GMT. I'm on the US Eastern seaboard so if I pick 5 May 2021, the output for me is 2021-050-05 04:00:00 +0000 since I'm about 4 hours off of GMT (I'm in our daylight savings time) and I set the time on the date to be 00:00:00. Basically, the date time is supposed to be GMT so you can convert it to local time.

In the MDPModel.swift, line 253:

```
let realDate = calendar.date(from: DateComponents(year: year, month: month, day: i+1))
```

This is the actual date behind the UI date circle on the screen. Notice that I'm not setting the time zone. You could add the time zone and try that:

```
let realDate = calendar.date(from: DateComponents(timeZone: TimeZone.current, year: year, month: month, day: i+1))
```

But add this to the ContentView.swift's `onChange(of: self.anyDays` function to see if you need to do that:

```
    for date in days {
                let formatter = DateFormatter()
                formatter.dateStyle = .short
                formatter.timeStyle = .none
                print("- \(date) formatted: \(formatter.string(from: date))")
     }
```

If the `formatted` date is what you expect, then you are good: the date being selected is correct.

**Comment by ravpninfotech** [comment]
It's done. Thank you 👍

**Comment by yoann-PMX** [comment]
Hi,
how can i init the calendar with a dateRange ?
